lib/dry/schema/extensions/json_schema/schema_compiler.rb

```diff
@@ -0,0 +1,182 @@
+# frozen_string_literal: true
+
+require "dry/schema/constants"
+
+module Dry
+  module Schema
+    # @api private
+    module JSONSchema
+      # @api private
+      class SchemaCompiler
+        IDENTITY = ->(v) { v }
+
+        PREDICATE_TO_TYPE = {
+          array?: {type: "array"},
+          bool?: {type: "boolean"},
+          date?: {type: "string", format: "date"},
+          date_time?: {type: "string", format: "date-time"},
+          decimal?: {type: "number"},
+          float?: {type: "number"},
+          hash?: {type: "object"},
+          int?: {type: "integer"},
+          nil?: {type: "null"},
+          str?: {type: "string"},
+          time?: {type: "string", format: "time"},
+          min_size?: {minLength: ->(v) { v.to_i }},
+          max_size?: {maxLength: ->(v) { v.to_i }},
+          included_in?: {enum: ->(v) { v.to_a }},
```

Comment:
Also AFAIU dry* prefers `:to_i.to_proc.freeze` and `IDENTITY` could be replaced by `:itself.to_proc.freeze`

Reply:
Unfortunately cant do this because the proc actually takes two args now